### PR TITLE
fix: skip invalid urls

### PR DESCRIPTION
**Pull request by markusahlstrand**
This was throwing before, but now it's skipping invalid urls

### CODE REVIEW

File: src/utils/validate-redirect-url.ts [hunk 1]
```diff
@@ -65,7 +65,8 @@ export function validateRedirectUrl(
     // const url = new URL(allowedUrl);
     const match: RegExpMatchArray | null = allowedUrl.match(urlPattern);
     if (!match) {
-      throw new Error("Invalid URL");
+      console.log(`Invalid URL: ${allowedUrl}`);
+      return null;
```
Comment (danielgent) [inline src/utils/validate-redirect-url.ts:69]: I did some PRs here if you want to go through them at some point https://github.com/sesamyab/auth/pull/178 https://github.com/sesamyab/auth/pull/177

I wrote comprehensive unit tests to show the current code is very permissive (will be a security hole if anyone else can edit these! ), and to not use regex :smile:

File: src/utils/validate-redirect-url.ts [hunk 2]
```diff
@@ -79,7 +80,7 @@ export function validateRedirectUrl(
   // Regular expression to remove trailing slashes, query strings, and fragments
   const cleanedUrl = redirectUri.replace(/(\/+)?(\?.*)?(#[^#]*)?$/, "");
 
-  if (!regexes.some((regex) => regex.test(cleanedUrl))) {
+  if (!regexes.some((regex) => regex?.test(cleanedUrl))) {
```
Comment (danielgent) [inline src/utils/validate-redirect-url.ts:83]: I worry about these changes that don't add unit tests :smile: 

But I'm not the one dealing with the support tickets at midnight Markus! Again, happy to go through a load of my open PRs and discuss how to improve your life quality :smile:

**Review by danielgent — APPROVED** [review]
This looks fine to my human eyes :mag:

File: src/utils/validate-redirect-url.ts (hunk 1 above)
Comment (markusahlstrand) [reply src/utils/validate-redirect-url.ts:69]: We escape all regex parameters so think it should be fine? 
```
function escapeRegExp(string) {
  return string.replace(/[.*+?^${}()|[\]\\/]/g, "\\$&");
}
```

File: src/utils/validate-redirect-url.ts (hunk 2 above)
Comment (markusahlstrand) [reply src/utils/validate-redirect-url.ts:83]: Yepp, should really add a test...

**Pull request merged**